tracing-appender/src/lib.rs

```diff
@@ -42,7 +107,61 @@ mod worker;
 /// Unintentional drops of `WorkerGuard` remove the guarantee that logs will be flushed
 /// during a program's termination, in a panic or otherwise.
 ///
+/// See [`WorkerGuard`][worker_guard] for examples of using the guard.
+///
+/// [worker_guard]: ./struct.WorkerGuard.html
+///
 /// # Examples
+///
+/// ``` rust
+/// # fn docs() {
+/// let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
+/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+/// tracing::subscriber::with_default(subscriber.finish(), || {
+///    tracing::event!(tracing::Level::INFO, "Hello");
+/// });
+/// # }
+/// ```
+pub mod non_blocking;
+
+/// A rolling file appender.
+///
+/// Creates a log file at a specified directory and file name prefix which *may* be appended with
+/// the date and time.
+///
+/// The following helpers are available for creating a rolling file appender.
+///
+/// - [`Rotation::hourly()`][hourly]:
+///     <br/> This will result in log file located at `some_directory/log_file_name_prefix.YYYY-MM-DD-HH`
+/// - [`Rotation::daily()`][daily]:
+///     <br/> This will result in log file located at `some_directory/log_file_name_prefix.YYYY-MM-DD`
+/// - [`Rotation::never()`][never]:
+///     <br/> This will result in log file located at `some_directory/log_file_name`
```

Comment:
I don't think the explicit line breaks are necessary?
```suggestion
/// - [`Rotation::hourly()`][hourly]: This will result in log file located at `some_directory/log_file_name_prefix.YYYY-MM-DD-HH`
/// - [`Rotation::daily()`][daily]: This will result in log file located at `some_directory/log_file_name_prefix.YYYY-MM-DD`
/// - [`Rotation::never()`][never]: This will result in log file located at `some_directory/log_file_name`
```
Also, I think it might be worth explaining that the rotation doesn't just control how the timestamps are formatted, but how frequently new log files are created as well?